core/src/banking_stage.rs

```diff
@@ -75,33 +76,131 @@ impl BankingStage {
 
     fn forward_unprocessed_packets(
         socket: &std::net::UdpSocket,
-        tpu: &std::net::SocketAddr,
-        unprocessed_packets: UnprocessedPackets,
+        forwarder: &std::net::SocketAddr,
+        unprocessed_packets: &[(SharedPackets, usize)],
     ) -> std::io::Result<()> {
+        let mut blob = Blob::default();
         for (packets, start_index) in unprocessed_packets {
             let packets = packets.read().unwrap();
-            for packet in packets.packets.iter().skip(start_index) {
-                socket.send_to(&packet.data[..packet.meta.size], tpu)?;
+            let mut current_index = *start_index;
+            while current_index < packets.packets.len() {
+                current_index += blob.store_packets(&packets.packets[current_index..]) as usize;
+                if current_index < packets.packets.len() {
+                    // Blob is full, send it
+                    socket.send_to(&blob.data[..blob.meta.size], forwarder)?;
```

Comment:
How about writing a `packets_to_blobs` function and then sending the blobs?

Reply:
Because the unprocessed packet queue size is unpredictable, I thought it might be better to send the blobs out on demand rather than constructing them all before sending them.

Reply:
Premature optimization that makes unit-test harder?

Reply:
Factored it into a separate function and added a test :D